docs/decisions/0014-single-taxonomy-view-api.rst

```diff
@@ -0,0 +1,156 @@
+
+**Branch:** Is the representation of a root Tag and all its children up to the leaves.
+
+For the decisions, the following use cases were taken into account:
+
+- List of root tags that can be expanded to show children tags.
+    - This list can be sorted alphabetically: A-Z (default) and Z-A
+- The user can expand all root tags.
+- The user can search for tags.
+
+
+Decision
+---------
+
+
+Tag representation
+~~~~~~~~
+
+Return a list of root tags and within each one have the list of children tags. Each root would have
+its entire branch. The list of root tags will be ordered alphabetically as is each listing
+at each level of the tree. This order can only be reversed in the root tag listing.
+
+{
+    "count": 100,
+    "tags": [
+        {
+            "id": "tag_1",
+            "value": "Tag 1",
+            "taxonomy_id": "1",
+            "sub_tags": [
+                {
+                    "id": "tag_2",
+                    "value": "Tag 2",
+                    "taxonomy_id": "1",
+                    "sub_tags": [
+                        (....)
+                    ]
+                },
+                (....)
+            ]
+        },
+        (....)
+    ]
+}
+
+
+**Pros:**
+
+- The edX's interfaces show the tags in the form of a tree.
+- The frontend needs no further processing as it is in a displayable format.
+- It is kept as a simple implementation.
+
+**Cons:**
+
+- More implementation on the API side.
+
+
+Pagination
+~~~~~~~~~~~
+
+Apply the pagination only in the root tags and bring the entire branch of each root.
+Children do not affect pagination in any way.
+
+**Pros**
+
+- It is the simplest way.
+
+**Cons**
+
+- The children would not have pagination, in the long run there may be cases in which
+  the branch has hundreds of children, and they would still all be brought.
+
+
+Search tags
+~~~~~~~~~~~~
+
+Support tag search on the backend. Return a subset of matching tags in the format proposed
+in this document.
+
+**Pros**
+
+- It is the most scalable way.
+
+
+Rejected Options
+-----------------
+
+
+Render as a simple list of tags
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Return a simple list of tags, regardless of whether it is root or leaf.
+
+**Pros:**
+
+- It is simple and does not need further implementation and processing in the API.
+
+**Cons:**
+
+- It is more work to re-process all that list in the frontend to know who it is whose father.
+- In no edX's interface is it used this way and it would be a very specific use case.
+- Pagination would be more complicated to perform.
+
+
+
+Get the branch in another call
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+Get the root tags in one call and all children tags of a branch in another call.
+This second function is called when the user expands the parent tag.
+
+**Cons:**
+
+- In the UI there is the functionality *Expand all*, another view would have to 
+  be made to handle this functionality in a scalable way.
```

Comment:
I personally like the "Get the branch in another call" approach the most. It's simple: first load only the first page of root tags, and then the user can load additional pages of root tags or expand any root tag and load [the first page of] its children. And so on, recursively.

If the main con is
> In the UI there is the functionality *Expand all*, another view would have to  be made to handle this functionality in a scalable way.
I think we can easily work around that by disabling the "Expand all" option for taxonomies that have > 1,000 tags.

Likewise, for taxonomies with < 1,000 tags we can pre-load all the data from the whole taxonomy.

But don't go changing the ADR just yet - let's ask on Slack for input.

Reply:
I [asked](https://www.figma.com/file/DmUC2e7BAoQsGDq4KgApPa?node-id=1878:29199&mode=design#533948123) on Figma.